src/plugins/augeas/augeas.c

```diff
@@ -111,12 +111,17 @@ int elektraAugeasGenConf (KeySet * ks, Key * errorKey ELEKTRA_UNUSED)
 	return retval;
 }
 
-static const char * getAugeasError (augeas * augeasHandle)
+static const char * getAugeasError (augeas * augeasHandle, const char * lensPath)
 {
 	const char * reason = 0;
 	if (aug_error (augeasHandle) != 0)
 	{
+		const char * format = "%s\n\tlensPath: %s";
 		reason = aug_error_message (augeasHandle);
+		size_t messageSize = strlen (reason) + strlen (lensPath) + strlen (format);
```

Comment:
I have copied this from here: https://github.com/ElektraInitiative/libelektra/blob/e74733627485950e31eb37c7af4badacbce7352f/src/plugins/augeas/augeas.c#L143-L150

This bug should then also appear there

Reply:
> This bug should then also appear there

I think you are right. The off-by-one error is also one of the reasons why `elektraStrLen` also counts the terminating `NULL` character.

https://github.com/ElektraInitiative/libelektra/blob/f3460ca7bf4371fbde2eb358e5217c9c658c8314/src/libs/elektra/internal.c#L369-L371

Reply:
@kodebach can you take a look here?

Reply:
What exactly should I look at?

@sanssecours is right in general, but actually the `+ 1` is not needed here. The string `format` is a `printf`-string in the resulting buffer the `%s` will not be present, but they are counted by `strlen`. So actually the allocated buffer is 3 bytes bigger than it needs to be.

I didn't check the second one. It might be more problematic, because in the `NULL` cases, `strlen` is replaced by 0, but the NULL string is replaced by `" "`. I think it still works, but adding 1 to the buffer size certainly doesn't hurt.

PS If someone is working on `augeas`, there are code paths where it sets an error, but continues to execute and in the end returns `SUCCESS`.

Reply:
> The string `format` is a `printf`-string in the resulting buffer the `%s` will not be present, but they are counted by `strlen`.

I did not think about that 😊. Thank you for the detailed explanation.